internal/component/otelcol/connector/connector.go

```diff
@@ -177,22 +196,44 @@ func (p *Connector) Update(args component.Arguments) error {
 	var metricsConnector otelconnector.Metrics
 	var logsConnector otelconnector.Logs
 
-	switch pargs.ConnectorType() {
-	case ConnectorTracesToMetrics:
+	ct := pargs.ConnectorType()
+	if ct.SupportsAnyToMetrics() {
```

Comment:
In theory a connector can be both AnyToMetrics and AnyToLogs, in which case we'd have to remove the `"this connector can only output metrics"` error below. But we can leave the code as it is for now... and think about those other cases as they arise. Flow is flexible enough so that we can rework our code in the future to accommodate other connector types.

Reply:
Yes, this makes sense. I thought about it as I was implementing this but decided not to tackle it yet as there isn't a use case for it yet.